src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/BookingIntoPrisonService.kt

```diff
@@ -93,8 +99,8 @@ class BookingIntoPrisonService(
     val bookNumber: String = bookNumberGenerationService.generateBookNumber()
     val staff = getLoggedInStaff().getOrThrow().staff
 
-    // now increment the sequence on each booking
-    offender.bookings.forEach(OffenderBooking::incBookingSequence)
+    // now increment the sequence on each booking - bookings are linked via the root offender
+    offender.rootOffender.bookings.forEach(OffenderBooking::incBookingSequence)
```

Comment:
`offender.bookings` incorrectly joins between the `offender.id` and `bookings.rootOffenderId`, which only finds any bookings if the `offender` is the root.  We therefore need to get the root first here